more_itertools/more.py

```diff
@@ -1064,28 +1064,43 @@ def sliced(seq, n):
     return takewhile(len, (seq[i : i + n] for i in count(0, n)))
 
 
-def split_at(iterable, pred):
+def split_at(iterable, pred, maxsplit=-1):
     """Yield lists of items from *iterable*, where each list is delimited by
-    an item where callable *pred* returns ``True``. The lists do not include
-    the delimiting items.
+    an item where callable *pred* returns ``True``.
+    The lists do not include the delimiting items:
 
         >>> list(split_at('abcdcba', lambda x: x == 'b'))
         [['a'], ['c', 'd', 'c'], ['a']]
 
         >>> list(split_at(range(10), lambda n: n % 2 == 1))
         [[0], [2], [4], [6], [8], []]
+
+    At most *maxsplit* splits are done. If *maxsplit* is not specified or -1,
+    then there is no limit on the number of splits:
+
+        >>> list(split_at(range(10), lambda n: n % 2 == 1, maxsplit=2))
+        [[0], [2], [4, 5, 6, 7, 8, 9]]
     """
+    if maxsplit == 0:
```

Comment:
Could you make this `if not maxsplit` so we can catch `None` also?

Reply:
I tested this:

```
>>> "1,2".split(",", maxsplit=0)
['1,2']
>>> "1,2".split(",", maxsplit=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object cannot be interpreted as an integer
```

I think we should stick to `str.split` behavior, and check the type of `maxsplit` like this:

```
>>> maxsplit = 0
>>> if not isinstance(maxsplit, int): raise TypeError("'{}' object cannot be interpreted as an integer".format(maxsplit.__class__.__name__))
... 
>>> maxsplit = None
>>> if not isinstance(maxsplit, int): raise TypeError("'{}' object cannot be interpreted as an integer".format(maxsplit.__class__.__name__))
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object cannot be interpreted as an integer
```

Reply:
Good call. Thanks!